framework/src/base/FEProblem.C

```diff
@@ -2948,6 +3088,30 @@ FEProblem::setCouplingMatrix(CouplingMatrix * cm)
   _cm = cm;
 }
 
+void
+FEProblem::setNonlocalCouplingMatrix()
+{
+  unsigned int n_vars = _nl.nVariables();
+  _nonlocal_cm = new CouplingMatrix(n_vars);
```

Comment:
We might just make `_nonlocal_cm` of type `CouplingMatrix` directly and initialize it to size zero. Then we can deal with straight up references (no ugly pointers). Here we would just call `_nonlocal_cm.resize(n_vars);`